### PR TITLE
Feature/client streaming

### LINKED ISSUE #13: implement client streaming calls

### PR DESCRIPTION
**Pull request by vadimi**
Finally client streaming support. I've refactored quite a bit on the input reader side to get ready for bi-directional streaming.

Closes #13

### CODE REVIEW

File: cmd/grpc-client-cli/app.go [hunk 1]
```diff
@@ -142,8 +153,10 @@ func (a *app) callService(method *desc.MethodDescriptor, message []byte) error {
 			err = a.callServerStream(ctx, method, selectedMsg)
 		} else if !method.IsServerStreaming() && !method.IsClientStreaming() {
 			err = a.callUnary(ctx, method, selectedMsg)
+		} else if !method.IsServerStreaming() && method.IsClientStreaming() {
+			err = a.callClientStream(ctx, method, messages)
 		} else {
-			err = errors.New("client/bi-directional streaming is not supported")
+			err = errors.New("bi-directional streaming is not supported")
 		}
```
Comment (Mainian) [inline cmd/grpc-client-cli/app.go:160]: Technically we check `!method.IsServerStreaming()` twice? But also use two different styles (lines 137 vs this)

Maybe nest the seccond check `!method.isClientStreaming()` and `method.isClientStreaming()`?

File: cmd/grpc-client-cli/msgbuffer.go [hunk 2]
```diff
@@ -0,0 +1,135 @@
...
+		// it means no new messages are expected
+		if err == io.EOF {
+			return buf, nil
+		}
+
+		if err != nil {
+			return nil, err
+		}
+
+		buf = append(buf, msg)
```
Comment (Mainian) [inline cmd/grpc-client-cli/msgbuffer.go:93]: there is a reader buffer size in msgreader.go 

Os there any size constraint we need to consider in the buffer ?

File: cmd/grpc-client-cli/app.go (hunk 1 above)
Comment (vadimi) [reply cmd/grpc-client-cli/app.go:164]: ok, check now, restructure a bit

File: cmd/grpc-client-cli/msgbuffer.go (hunk 2 above)
Comment (vadimi) [reply cmd/grpc-client-cli/msgbuffer.go:93]: this buffer is used to buffer messages for input stream (client or bi-directional streaming), I don't expect anyone would manually add so many messages that the app would run out of memory

**Pull request merged**